src/field_theory/ratpoly.lean

```diff
@@ -0,0 +1,391 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import ring_theory.localization
+
+/-!
+# The field of rational polynomials
+
+This file defines the field `ratpoly K` of rational polynomials over a field `K`,
+and shows it is the field of fractions of `polynomial K`.
+
+## Implementation notes
+
+We need a couple of maps to set up the `field` and `is_fraction_ring` structure,
+namely `ratpoly.of_fraction_ring`, `ratpoly.to_fraction_ring`, `ratpoly.mk` and `ratpoly.aux_equiv`.
+All these maps get `simp`ed to bundled morphisms like `algebra_map (polynomial K) (ratpoly K)`
+and `is_localization.alg_equiv`.
+-/
+
+noncomputable theory
+open_locale classical
+open_locale non_zero_divisors
+
+universes u v
+
+variables (K : Type u) [field K]
```

Comment:
Should the definition be made more generally then fields?

Reply:
Everything in the file generalizes for free if `K` is an integral domain. Although `ratpoly A` would be isomorphic to `ratpoly (fraction_ring A)`, so I'm not sure how much we gain.

Reply:
Sure, we can always generalize later. But I can imagine that it's useful to sometimes plug `ℤ` into this construction. Let's leave it like this for now.